CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptPkcs7VerifyCommon.c

```diff
@@ -541,7 +540,10 @@ MbedtlsPkcs7SignedDataVerifySigners (
   HashLen = mbedtls_md_get_size(MdInfo);
   mbedtls_md (MdInfo, Data, DataLen, Hash);
   if (SignerInfo->AuthAttr.p != NULL) {
+    TempAuthAttr = *(SignerInfo->AuthAttr.p);
+    *(SignerInfo->AuthAttr.p) = 0x31;
     mbedtls_md (MdInfo, SignerInfo->AuthAttr.p, SignerInfo->AuthAttr.len, Hash);
+    *(SignerInfo->AuthAttr.p) = TempAuthAttr;

```

Comment:
please explain why we need restore?

Reply:
Thanks. Because if we don't restore, the `MbedtlsPkcs7SignedDataVerifySigners` will change the signerinfo.
And the calling sequence is:  `AuthenticodeVerify -> Pkcs7Verify-> MbedTlsPkcs7SignedDataVerify->MbedtlsPkcs7SignedDataVerifySigners`, which will change the SignedData. And the SignedData will be used in other place: such as Pkcs7GetSigners.

```
    if (AuthenticodeVerify (SignedData, SignedDataSize, RevokedCert, RevokedCertSize, InHash, InHashSize)) {
      //
      // The signedData was verified by one entry in Revoked Database
      //
      Status = EFI_SUCCESS;
      break;
    }
  }

  if (!EFI_ERROR (Status)) {
    //
    // The signedData was revoked, since it was hit by RevokedDb
    //
    goto _Exit;
  }

  //
  // Now we will continue to check the X.509 Certificate Hash & Possible Timestamp
  //
  if ((TimeStampDb == NULL) || (*TimeStampDb == NULL)) {
    goto _Exit;
  }

  Pkcs7GetSigners (SignedData, SignedDataSize, &CertBuffer, &BufferLength, &TrustedCert, &TrustedCertLength);
```